File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> stk;
    int i = 0;

    while (i < arr.size()) {
        if (stk.empty()) {
            stk.push_back(arr[i]);
            i++;
        } else {
            if (stk.back() == arr[i]) {
                stk.pop_back();
                i++;
            } else {
                stk.push_back(arr[i]);
                i++;
            }
        }
    }

    if(stk.empty()) {
        stk.push_back(-1);
    }

    return stk;
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.swift
import Foundation

func solution(_ arr:[Int]) -> [Int] {
    
    var stk: [Int] = []
    var i = 0
    
    while i < arr.count {
        if stk.isEmpty {
            stk.append(arr[i])
        } else {
            (stk.last == arr[i]) ? _ = stk.removeLast() : stk.append(arr[i])
        }
        i += 1
    }
    
    return !stk.isEmpty ? stk : [-1]
}